src/3rd_party_add.c

```diff
@@ -77,12 +77,18 @@ enum swupd_code third_party_add_main(int argc, char **argv)
 	}
 	progress_init_steps("third-party", step_in_third_party_add);
 
-	ret = swupd_init(SWUPD_NO_ROOT);
+	ret = swupd_init(SWUPD_ALL);
 	if (ret != SWUPD_OK) {
 		goto finish;
 	}
 
-	/* TODO implement */
+	/* The last two in reverse are the repo-name, repo-url */
+	if (add_repo_config(argv[argc - 2], argv[argc - 1])) {
```

Comment:
maybe we should validate that the URL format is valid (that it begins with http:// or https:// or file://)

Reply:
Thanks will change in next push